torch/package/package_importer.py

```diff
@@ -178,49 +271,17 @@ def load_pickle(self, package: str, resource: str, map_location=None) -> Any:
             Any: The unpickled object.
         """
         pickle_file = self._zipfile_path(package, resource)
-        restore_location = _get_restore_location(map_location)
-        loaded_storages = {}
         loaded_reduces = {}
-        storage_context = torch._C.DeserializationStorageContext()
-
-        def load_tensor(dtype, size, key, location, restore_location):
-            name = f"{key}.storage"
-
-            if storage_context.has_storage(name):
-                storage = storage_context.get_storage(name, dtype).storage()
-            else:
-                tensor = self.zip_reader.get_storage_from_record(
-                    ".data/" + name, size, dtype
-                )
-                if isinstance(self.zip_reader, TorchScriptPackageZipFileReader):
-                    storage_context.add_storage(name, tensor)
-                storage = tensor.storage()
-            loaded_storages[key] = restore_location(storage, location)
 
         def persistent_load(saved_id):
             assert isinstance(saved_id, tuple)
             typename = _maybe_decode_ascii(saved_id[0])
             data = saved_id[1:]
-
-            if typename == "storage":
-                storage_type, key, location, size = data
-                dtype = storage_type.dtype
-
-                if key not in loaded_storages:
-                    load_tensor(
-                        dtype,
-                        size,
-                        key,
-                        _maybe_decode_ascii(location),
-                        restore_location,
-                    )
-                storage = loaded_storages[key]
-                # TODO: Once we decide to break serialization FC, we can
-                # stop wrapping with _TypedStorage
-                return torch.storage._TypedStorage(
-                    wrap_storage=storage._untyped(), dtype=dtype
-                )
-            elif typename == "reduce_package":
+            if self.torch_is_available:
+                module = self.persistent_id_torch(typename, data)
```

Comment:
same thing here, we could have two loader methods with identical interfaces that are selected at import time, that way we can cut down on these if else checks